public/js/positionList.js

```diff
@@ -0,0 +1,83 @@
+const openEditPosition = (id) => {
+  window.location.href = `${window.location.origin}/views/positionForm.html?_id=${id}`;
+};
+
+const openNewPosition = () => {
+  window.location.href = `${window.location.origin}/views/positionForm.html`;
+};
+
+window.onload = () => {
+  const navButton = document.getElementById('positionsNav');
+  navButton.classList.add('activePage');
+
+  const tableContent = document.getElementById('positions-table-content');
+
+  const addPosition = document.getElementById('addPosition');
+  addPosition.onclick = openNewPosition;
+
+  const modalSection = document.getElementById('modal-section');
+  const openModal = () => {
+    modalSection.classList.add('modal-section-display-on');
+  };
+
+  const closeModal = (e) => {
+    if (e.target === modalSection) {
+      modalSection.classList.remove('modal-section-display-on');
+      modalSection.classList.add('modal-display-off');
+    }
+  };
+
+  window.addEventListener('click', closeModal);
+  const modalCloseBtn = document.getElementById('modal-close-button');
+  modalCloseBtn.onclick = () => {
+    modalSection.classList.remove('modal-section-display-on');
+    modalSection.classList.add('modal-display-off');
+  };
+
+  const deletePosition = (id, jobDescription, event) => {
+    event.stopPropagation();
+    document.getElementById('modal-title').innerText = 'You are about to delete the open Position of';
+    document.getElementById('modal-data-inputs').innerText = `${jobDescription}`;
+    openModal();
+    document.getElementById('cancel-button').onclick = () => {
+      modalSection.classList.remove('modal-section-display-on');
+      modalSection.classList.add('modal-display-off');
+    };
+    document.getElementById('procced-button').onclick = () => {
+      const url = `${window.location.origin}/api/positions/${id}`;
+      fetch(url, {
+        method: 'DELETE',
+        headers: {
+          'Content-type': 'application/json',
+        },
+      })
+        .then(window.location.reload())
```

Comment:
you should check if the deletion was ok before redirecting the page. Anyway, you should close the modal and get the new position list instead of reloading the page.

Reply:
This can be resolved with this promise?
` .then((response) => {
        response.json();
        window.location.href = `${window.location.origin}/views/sessionList.html`;
      })
      .catch((error) => error);
  };
````

Reply:
.then((response) => {
        response.json();
        window.location.href = `${window.location.origin}/views/sessionList.html`;
      })
      .catch((error) => error);
  };

Reply:
done!